cmd/pulumi-resource-docker-buildkit/main.go

```diff
@@ -274,8 +274,9 @@ func (k *dockerBuildkitProvider) dockerBuild(
 		"--target", target,
 		"-t", name, "--push",
 	}
-	buildArgs := []map[string]string{}
+	var buildArgs []map[string]string
 	if !inputs["args"].IsNull() {
+		buildArgs = []map[string]string{}
```

Comment:
This looks unnecessary! `append` should automatically allocate if given a `nil` input.

```suggestion
```

Reply:
oh, i think i was trying to make sure that it was an empty slice rather than nil in this case, but i think it was doing that anyway because of something about the way it was doing reflection on these values. let me do a bit more testing.

Reply:
Ohhh, I see. So you want it to be nil if null input, or empty slice if empty, or non-empty slice if non-empty. Ok, yeah, totally checks out—assuming that is actually the requirement!

Reply:
i think the code ends up treating a nil slice as an empty slice anyway, so that line is actually unnecessary